File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        const char red = 0;
        const char white = 1;
        const char blue = 2;

        int l = 0;
        int m = 0;
        int r = nums.size() - 1;

        while(m <= r){
            switch(nums[m]){
                case red:
                    swap(nums[l],nums[m]);
                    l++;
                    m++;
                    break;
                case white :
                    m++;
                    break;
                case blue :
                    swap(nums[m], nums[r]);
                    r--;
                    break;
            }
        }
    }
};

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        n = len(nums)
        left = 0
        mid = 0
        right = n-1

        while mid <= right:
            if nums[mid] == 0:
                nums[left], nums[mid] = nums[mid], nums[left]
                left += 1
                mid += 1
            elif nums[mid] == 1:
                mid += 1
            else:
                nums[mid] , nums[right] = nums[right], nums[mid]
                right -= 1


        